### PR TITLE
update

### PR DESCRIPTION
**Pull request by michal-dagan**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
https://jira-dc.paloaltonetworks.com/browse/CIAC-16562
## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by marketplace-ai-reviewer** [comment]
🤖 Analysis started. Please wait for results...

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py [hunk 1]
```diff
@@ -1710,7 +1711,7 @@ def upload_file_command(client: Client, args: dict) -> tuple[str, dict, dict, bo
     file_name = args.get("file_name")
     if not file_name:
         file_data = demisto.getFilePath(file_id)
-        file_name = file_data.get("name")
+        file_name = os.path.basename(file_data.get("name"))
```
Comment (marketplace-ai-reviewer) [inline Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py]: Potential `TypeError` if `name` is missing from `file_data`.

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py [hunk 2]
```diff
@@ -3372,7 +3373,7 @@ def update_remote_system_with_entries(client, entries, params, ticket_id, ticket
         # Mirroring files as entries
         if is_entry_type_mirror_supported(entry.get("type")):
             path_res = demisto.getFilePath(entry.get("id"))
-            full_file_name = path_res.get("name")
+            full_file_name = os.path.basename(path_res.get("name"))
```
Comment (marketplace-ai-reviewer) [inline Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py]: Potential `TypeError` if `name` is missing from `path_res`.

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2_test.py [hunk 3]
```diff
@@ -4337,3 +4337,41 @@ def test_create_item_order_no_validation_true_adds_flag(self, mocker):
...
+def test_upload_file_command_uses_basename(mocker):
+    """
+    Given:
+        - A file entry with a name containing directory path components and no explicit file_name arg.
+    When:
+        - Calling upload_file_command.
+    Then:
+        - Verify that only the basename of the file name is used.
+    """
+    import os
```
Comment (marketplace-ai-reviewer) [inline Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2_test.py]: Consider moving this import to the top of the file.

File: Packs/ServiceNow/ReleaseNotes/2_9_7.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+
+##### ServiceNow v2
+- Improved filename handling during file upload operations.
```
Comment (marketplace-ai-reviewer) [inline Packs/ServiceNow/ReleaseNotes/2_9_7.md]: The release note entry does not start with an approved prefix. Please use one of the approved prefixes, such as `Improved implementation` or `Fixed an issue`.

For example:
```markdown
- Improved implementation of filename handling during file upload operations.
```

**Standard RN Fix Guidance:**
Please auto-generate the release notes via `demisto-sdk update-release-notes` and/or refer to the [Release Notes Documentation](https://xsoar.pan.dev/docs/documentation/release-notes#general) for the complete list of approved prefixes.

**Comment by marketplace-ai-reviewer** [comment]
<details>
<summary><strong>🤖 AI Review Disclaimer</strong></summary>

This review was generated by an AI-powered tool and may contain inaccuracies. Please be advised, and we extend our sincere apologies for any inconvenience this may cause.
</details>

**Review by marketplace-ai-reviewer — COMMENTED** [review]
Thanks for your contribution to the ServiceNow integration! I've reviewed the PR and left a few notes to help polish the code. 

Mainly, please guard against potential `TypeError`s when accessing the `name` key during file operations, and ensure your release notes start with an approved prefix. There are also a couple of minor test file adjustments to make, such as moving an import to the top and adding a unit test for path sanitization. 

Let me know if you have any questions!

Additionally, please address the following file-level notes:
- `Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2_test.py`: Consider adding a unit test for the path sanitization in `update_remote_system_with_entries`.

 @inbalapt1, @dantavori, @michal-dagan please review and approve the results generated by the AI Reviewer by responding 👍 on this comment.

**Comment by michal-dagan** [comment]
Addressed all AI reviewer feedback:
- Added default value guard (`""`) to `file_data.get("name", "")` and `path_res.get("name", "")` to prevent potential `TypeError`
- Moved `import os` to top of test file
- Added `test_update_remote_system_with_entries_uses_basename` unit test for path sanitization in `update_remote_system_with_entries`
- Updated release note in `2_9_7.md` to use approved prefix: 'Fixed an issue with filename handling during file upload operations.'

@marketplace-ai-reviewer re-review

**Comment by marketplace-ai-reviewer** [comment]
🤖 Analysis started. Please wait for results...

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2_test.py [hunk 5]
```diff
@@ -4337,3 +4338,77 @@ def test_create_item_order_no_validation_true_adds_flag(self, mocker):
...
+            "download_link": "https://test.com/download",
+            "sys_id": "abc123",
+        }
+    }
+
+    entries = [{"id": "entry_id", "type": 3, "tags": []}]
+    params = {
+        "file_tag_from_service_now": "FileFromServiceNow",
+        "file_tag": "file",
+    }
```
Comment (marketplace-ai-reviewer) [inline Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2_test.py:4403]: `TypeError` due to incorrect number of arguments.

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2_test.py [hunk 6]
```diff
@@ -4337,3 +4338,77 @@ def test_create_item_order_no_validation_true_adds_flag(self, mocker):
...
+    update_remote_system_with_entries(mock_client, entries, params, "ticket_id", "incident", {})
+
+    call_args = mock_client.upload_file.call_args[0]
+    full_file_name_used = call_args[2]
```
Comment (marketplace-ai-reviewer) [inline Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2_test.py]: Explicitly assert the expected file name.

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py [hunk 7]
```diff
@@ -1710,7 +1711,7 @@ def upload_file_command(client: Client, args: dict) -> tuple[str, dict, dict, bo
     file_name = args.get("file_name")
     if not file_name:
         file_data = demisto.getFilePath(file_id)
-        file_name = file_data.get("name")
+        file_name = os.path.basename(file_data.get("name", ""))
```
Comment (marketplace-ai-reviewer) [inline Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py]: Use `or ""` to safely handle `None` values.

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py [hunk 8]
```diff
@@ -3372,7 +3373,7 @@ def update_remote_system_with_entries(client, entries, params, ticket_id, ticket
         # Mirroring files as entries
         if is_entry_type_mirror_supported(entry.get("type")):
             path_res = demisto.getFilePath(entry.get("id"))
-            full_file_name = path_res.get("name")
+            full_file_name = os.path.basename(path_res.get("name", ""))
```
Comment (marketplace-ai-reviewer) [inline Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py]: Use `or ""` to safely handle `None` values.

**Comment by marketplace-ai-reviewer** [comment]
<details>
<summary><strong>🤖 AI Review Disclaimer</strong></summary>

This review was generated by an AI-powered tool and may contain inaccuracies. Please be advised, and we extend our sincere apologies for any inconvenience this may cause.
</details>

**Review by marketplace-ai-reviewer — COMMENTED** [review]
#### 🔄 Re-reviewing updates...
> [!IMPORTANT]
> To maintain review integrity, threads marked as **Resolved** without accompanying changes or explanations will be automatically **unresolved** for further verification.

Hi there! Thanks for your contribution to the ServiceNowv2 integration. 

I've left a few notes for you to look at, primarily around safely handling `None` values for file names using `or ""` in the main code. On the testing side, please fix a `TypeError` related to argument counts, explicitly assert expected file names, and add coverage for missing file names. 

Let me know if you have any questions!

Additionally, please address the following file-level notes:
- `Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2_test.py`: Add test coverage for missing or `None` file names.

 @inbalapt1, @dantavori, @marketplace-ai-reviewer, @michal-dagan please review and approve the results generated by the AI Reviewer by responding 👍 on this comment.

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2_test.py (hunk 5 above)
Comment (michal-dagan) [reply Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2_test.py:4403]: Fixed in the latest commit: corrected to 5-argument call matching the function signature `update_remote_system_with_entries(client, entries, params, ticket_id, ticket_type)`.

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2_test.py (hunk 6 above)
Comment (michal-dagan) [reply Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2_test.py]: Fixed in the latest commit: added explicit assertions `assert "passwd" in file_name_used`, `assert "/" not in file_name_used`, `assert ".." not in file_name_used`.

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py (hunk 7 above)
Comment (michal-dagan) [reply Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py]: Fixed in the latest commit: changed to `os.path.basename(file_data.get("name") or "")` to safely handle None values.

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py (hunk 8 above)
Comment (michal-dagan) [reply Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py]: Fixed in the latest commit: changed to `os.path.basename(path_res.get("name") or "")` to safely handle None values.

**Comment by marketplace-ai-reviewer** [comment]
🤖 Analysis started. Please wait for results...

**Comment by marketplace-ai-reviewer** [comment]
<details>
<summary><strong>🤖 AI Review Disclaimer</strong></summary>

This review was generated by an AI-powered tool and may contain inaccuracies. Please be advised, and we extend our sincere apologies for any inconvenience this may cause.
</details>

**Review by marketplace-ai-reviewer — COMMENTED** [review]
#### 🔄 Re-reviewing updates...
> [!IMPORTANT]
> To maintain review integrity, threads marked as **Resolved** without accompanying changes or explanations will be automatically **unresolved** for further verification.

Thank you for your contribution! Re-reviewed your PR, No new issues found. Great work! 👍 
Please check the previous comments to ensure all feedback has been resolved.

 @dantavori, @inbalapt1, @michal-dagan, @marketplace-ai-reviewer please review and approve the results generated by the AI Reviewer by responding 👍 on this comment.

**Review by inbalapt1 — APPROVED** [review]
Checked.

**Comment by marketplace-ai-reviewer** [comment]
🤖 Analysis started. Please wait for results...

**Comment by marketplace-ai-reviewer** [comment]
<details>
<summary><strong>🤖 AI Review Disclaimer</strong></summary>

This review was generated by an AI-powered tool and may contain inaccuracies. Please be advised, and we extend our sincere apologies for any inconvenience this may cause.
</details>

**Review by marketplace-ai-reviewer — COMMENTED** [review]
Thanks for submitting your PR!
Before moving on with the detailed review, please take a moment to address the following general comments.


- The PR title and description are too generic ('update') and do not reflect the actual changes. Please update the title and description to explicitly state that this PR fixes a path traversal issue by using `os.path.basename` for file names in the ServiceNowv2 integration.

**Comment by inbalapt1** [comment]
This PR requires Force Merge because of failing TPBs that are not related to the changes here. Attaching an empty change PR with the same failures: [https://github.com/demisto/content/pull/43741](https://github.com/demisto/content/pull/43741)

**Review by dantavori — APPROVED** [review]
